ocaml/testsuite/tests/typing-unique/overwriting_underscore.ml

```diff
@@ -11,7 +11,7 @@ let underscore () = _
 Line 1, characters 20-21:
 1 | let underscore () = _
                         ^
-Error: Syntax error: "wildcard "_"" not expected.
+Error: Syntax error: "wildcard _" not expected.
```

Comment:
I hate to nitpick here, but I was hoping to remove the outer quotes, not the inner ones. This is what happens before your patch on this syntax. If it's hard to remove the outer quotes (I understand how it might be!), then I would just craft a better error message. Normally, the stuff in quotes is stuff from the user's code, but the word `wildcard` doesn't appear in the code.

Reply:
But the outer quotes were introduced by upstream: https://github.com/ocaml/ocaml/commit/f042991a8764d61c2155f737371a444dd2d1850f

Reply:
For that matter, upstream now has nested quotes in their error message, so arguably we should follow that: https://github.com/ocaml/ocaml/blob/9bb7929ce61f5cb4457d294f16dee3ee132bc9ef/parsing/parser.mly#L2449

Reply:
I don't observe nested quotes in the upstream error message. (Yes the quotes are nested in the source code, but not in the output.)

```
[igm-qws-u22468b] 09:44:27 ~/temp> cat bug.ml
let underscore () = _
[igm-qws-u22468b] 09:44:30 ~/temp> ocaml -version
The OCaml toplevel, version 5.3.0+dev4-2024-09-20
[igm-qws-u22468b] 09:44:34 ~/temp> ocamlc bug.ml
File "bug.ml", line 1, characters 20-21:
1 | let underscore () = _
                        ^
Error: Syntax error: wildcard "_" not expected.
```

Reply:
Ah, is the `wildcard "_"` part highlighted in your shell? I think ocaml might change the display depending on whether highlighting is available or not (and use quotes if not). My code also highlights the `wildcard _` when run in the shell.

Reply:
I have reverted the commit "Remove nested quotes" since it clearly does not do what we want. With the reverted commit, I get the same output on the shell in my fork and upstream.